src/resmom/linux/alps.c

```diff
@@ -3660,94 +3512,80 @@ inventory_loop_on_segments(basil_node_t *node, vnl_t *nv, char *arch,
 			 * in the name.
 			 */
 			snprintf(vname, sizeof(vname), "%s_%ld",
-				mpphost, node->node_id);
+				 mpphost, node->node_id);
 			vname[sizeof(vname)-1] = '\0';
 		}
 
 		attr = "sharing";
 		/* already exists so don't define type */
 		if (vn_addvnr(nv, vname, attr,
-			ND_Force_Exclhost,
-			0, 0, NULL) == -1)
+			      ND_Force_Exclhost,
+			      0, 0, NULL) == -1)
 			goto bad_vnl;
 
 		attr = "resources_available.PBScrayorder";
 		sprintf(utilBuffer, "%ld", order);
 		if (vn_addvnr(nv, vname, attr, utilBuffer,
-			ATR_TYPE_LONG, atype,
-			NULL) == -1)
+			      ATR_TYPE_LONG, atype,
+			      NULL) == -1)
 			goto bad_vnl;
 
 		attr = "resources_available.arch";
 		if (vn_addvnr(nv, vname, attr, arch,
-			0, 0, NULL) == -1)
+			      0, 0, NULL) == -1)
 			goto bad_vnl;
 
 		attr = "resources_available.host";
 		sprintf(utilBuffer, "%s_%ld", mpphost, node->node_id);
 		if (vn_addvnr(nv, vname, attr, utilBuffer,
-			0, 0, NULL) == -1)
+			      0, 0, NULL) == -1)
 			goto bad_vnl;
 
 		attr = "resources_available.PBScraynid";
 		sprintf(utilBuffer, "%ld", node->node_id);
 		if (vn_addvnr(nv, vname, attr, utilBuffer,
-			ATR_TYPE_STR, atype,
-			NULL) == -1)
+			      ATR_TYPE_STR, atype,
+			      NULL) == -1)
 			goto bad_vnl;
 
 		if (vnode_per_numa_node) {
 			attr = "resources_available.PBScrayseg";
 			sprintf(utilBuffer, "%d", totseg);
 			if (vn_addvnr(nv, vname, attr, utilBuffer,
-				ATR_TYPE_STR, atype,
-				NULL) == -1)
+				      ATR_TYPE_STR, atype,
+				      NULL) == -1)
 				goto bad_vnl;
 		}
 
 		attr = "resources_available.vntype";
 		if (vn_addvnr(nv, vname, attr, CRAY_COMPUTE,
-			0, 0, NULL) == -1)
+			      0, 0, NULL) == -1)
 			goto bad_vnl;
 
 		attr = "resources_available.PBScrayhost";
 		if (vn_addvnr(nv, vname, attr, mpphost,
-			ATR_TYPE_STR, atype,
-			NULL) == -1)
+			      ATR_TYPE_STR, atype,
+			      NULL) == -1)
 			goto bad_vnl;
 
 		if (vnode_per_numa_node) {
-			/* This is the default behavior */
-			attr = "resources_available.ncpus";
 			totcpus = 0;
-			for (proc = seg->processors; proc; proc = proc->next)
-				totcpus++;
-			sprintf(utilBuffer, "%d", totcpus);
-			if (vn_addvnr(nv, vname, attr, utilBuffer,
-				0, 0, NULL) == -1)
-				goto bad_vnl;
-
 			cu = seg->computeunits;
 			if (cu) {
-				attr = "resources_available.vps_per_ppu";
-				sprintf(utilBuffer, "%d", cu->proc_per_cu_count);
-				if (vn_addvnr(nv, vname, attr, utilBuffer,
-					0, 0, NULL) == -1)
-					goto bad_vnl;
-
-				totppus = 0;
 				for (cu = seg->computeunits; cu; cu = cu->next) {
-					totppus++;
-				}
-				if (totppus > 0) {
-					attr = "resources_available.nppus";
-					sprintf(utilBuffer, "%d", totppus);
-					if (vn_addvnr(nv, vname, attr, utilBuffer,
-						0, 0, NULL) == -1)
-						goto bad_vnl;
+					totcpus++;
```

Comment:
As explained in previous comment, that in basil 1.3 "Processor" elements does not represent actual cores, but the virtual cpus for each core. Therefore counting compute unit is right way to get ncpus.